endpoints/test/users_id_token_test.py

```diff
@@ -432,14 +432,14 @@ class DummyResponse(object):
       status_code = 200
       content = json.dumps(token)
 
-    self.mox.StubOutWithMock(urlfetch, 'fetch')
-    urlfetch.fetch(mox.IsA(basestring)).AndReturn(DummyResponse())
-
-    self.mox.ReplayAll()
-    users_id_token._set_bearer_user_vars_local('unused_token',
-                                               self._SAMPLE_ALLOWED_CLIENT_IDS,
-                                               self._SAMPLE_OAUTH_SCOPES)
-    self.mox.VerifyAll()
+    expected_uri = 'https://www.googleapis.com/oauth2/v1/tokeninfo?access_token=unused_token'
+    with mock.patch.object(urlfetch, 'fetch') as mock_fetch:
```

Comment:
Any reason this is not done as a decorator?

Reply:
Patching with decorators interacts weirdly with keyword args in certain conditions, because patched mock objects are appended to the positional args.

```python
@patch.object(foo, 'bar')
def MyFunc(self, mock_foo, example=None):
  pass

MyFunc(example=123)  # mock_foo is the mock, example is 123
```

```python
@patch.object(foo, 'bar')
def MyFunc(self, mock_foo, example=None):
  pass

MyFunc(123)  # mock_foo is 123, example is the mock
```

For this reason, it's best not to use patch as a function decorator except on functions which are not called directly by the test framework.